### PR TITLE
Update registration, transfer, renewal responses

### PR DESCRIPTION
**Pull request by weppos**
This PR update the response from a register/transfer/renew domain request, as these API calls are now return representation of a Transfer, Registration, and Renewal instead of a Domain.

See dnsimple/dnsimple-developer#111

### CODE REVIEW

**Comment by weppos** [comment]
This is now ready for review.

/cc @jodosha @aeden

File: dnsimple/registrar.go [hunk 1]
```diff
@@ -54,10 +54,24 @@ type DomainRegisterRequest struct {
...
+// DomainRegistration represents the result of a domain renewal call.
+type DomainRegistration struct {
+	ID           int    `json:"id"`
+	DomainID     int    `json:"domain_id"`
+	RegistrantID int    `json:"registrant_id"`
+	Period       int    `json:"period"`
+	State        string `json:"state"`
+	AutoRenew    bool   `json:"auto_renew"`
+	PrivateWhois bool   `json:"private_whois"`
+	PremiumPrice int    `json:"premium_price"`
```
Comment (jodosha) [inline dnsimple/registrar.go]: @weppos Shall this be a float?

File: dnsimple/registrar.go [hunk 2]
```diff
@@ -147,10 +174,21 @@ type DomainRenewRequest struct {
 	Period int `json:"period"`
 }
 
-// DomainRenewalResponse represents a response from an API method that results in a domain renewal.
+// DomainRenewal represents the result of a domain renewal call.
+type DomainRenewal struct {
```
Comment (jodosha) [inline dnsimple/registrar.go]: @weppos This is missing `state`

File: dnsimple/registrar.go [hunk 3]
```diff
@@ -94,10 +108,23 @@ type DomainTransferRequest struct {
...
 
+// DomainTransfer represents the result of a domain renewal call.
+type DomainTransfer struct {
+	ID           int    `json:"id"`
+	DomainID     int    `json:"domain_id"`
+	RegistrantID int    `json:"registrant_id"`
+	State        string `json:"state"`
+	AutoRenew    bool   `json:"auto_renew"`
+	PrivateWhois bool   `json:"private_whois"`
+	PremiumPrice int    `json:"premium_price"`
```
Comment (jodosha) [inline dnsimple/registrar.go]: @weppos similar to https://github.com/dnsimple/dnsimple-go/pull/52/files#r91865332

File: dnsimple/registrar.go (hunk 1 above)
Comment (weppos) [inline dnsimple/registrar.go]: Good question. Is it a Float? I've just noticed in https://github.com/dnsimple/dnsimple-elixir/pull/91 that this seems to be a String.

Can you confirm what value it is, and if it is consistent?

File: dnsimple/registrar.go (hunk 2 above)
Comment (weppos) [inline dnsimple/registrar.go]: ✔️

**Comment by weppos** [comment]
@jodosha I made the changes accordingly

**Comment by jodosha** [comment]
👍

**Pull request merged**